packages/nightly/src/lib/nightly.ts

```diff
@@ -0,0 +1,175 @@
+import type {
+  InjectedWallet,
+  WalletBehaviourFactory,
+  WalletModuleFactory,
+} from "@near-wallet-selector/core";
+import { Account, waitFor } from "@near-wallet-selector/core";
+import { createAction } from "@near-wallet-selector/wallet-utils";
+import { isMobile } from "is-mobile";
+import { utils } from "near-api-js";
+import { AccessKeyView } from "near-api-js/lib/providers/provider";
+import { createTransaction } from "near-api-js/lib/transaction";
+import { PublicKey } from "near-api-js/lib/utils";
+import type { NearNightly, NightlyInjected } from "./injected-nightly";
+
+declare global {
+  interface Window {
+    nightly: NightlyInjected | undefined;
+  }
+}
+
+const setupNightlyState = (): NearNightly | undefined => {
+  return window.nightly?.near;
+};
+
+const Nightly: WalletBehaviourFactory<InjectedWallet> = async ({
+  metadata,
+  store,
+  logger,
+  provider,
+}) => {
+  const _state = setupNightlyState();
+  const currentState = store.getState();
+  if (currentState.selectedWalletId === "nightly") {
+    try {
+      // eager connect to the wallet
+      await _state?.connect(undefined, true);
+    } catch {
+      // ignore
+    }
+  }
```

Comment:
As mentioned previously, once we've addressed the installed status of the extension, we can assume `window.nightly.near` exists which will clean up the defensive approach that's be used (e.g. `?.`).

```suggestion
declare global {
  interface Window {
    nightly: InjectedNightly | undefined;
  }
}

interface NightlyState {
  wallet: InjectedNightly;
}

const setupNightlyState = async (store: WalletSelectorStore): NightlyState => {
  const { selectedWalletId } = store.getState();
  const wallet = window.nightly.near!
  
  // Attempt to reconnect wallet if previously selected.
  if (selectedWalletId === "nightly") {
      await wallet.connect(undefined, true).catch(() => null);
  }

  return {
    wallet,
  };
};

const Nightly: WalletBehaviourFactory<InjectedWallet> = async ({
  metadata,
  store,
  logger,
  provider,
}) => {
  const _state = await setupNightlyState(store);
```

Reply:
refactor here https://github.com/near/wallet-selector/pull/314/commits/33bb3202b47ffc3f0fe2dbc2fc8460ab56816baf